File: algorithms/0704.binary-search/solution.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let mut left = 0 as i32;
        let mut right = (nums.len() - 1) as i32;

        while left <= right {
            let mid = ((left + right) / 2) as usize;

            if target == nums[mid] {
                return mid as i32;
            } else if target < nums[mid] {
                right = mid as i32 - 1;
            } else {
                left = mid as i32 + 1;
            }
        }

        -1
    }
}

File: algorithms/0704.binary-search/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
  ## Using Array#bsearch_index
  # nums.bsearch_index {|x| target - x } || -1

  left, right = 0, nums.size - 1
  while left <= right
    mid = (left + right) / 2

    if target == nums[mid]
      return mid
    elsif target < nums[mid]
      right = mid - 1
    else
      left = mid + 1
    end
  end
  -1
end
